src/keyboard/mod.rs

```diff
@@ -435,6 +447,26 @@ pub enum Event<'a> {
     },
 }
 
+/// An event sent at repeated intervals for certain keys determined by xkb_keymap_key_repeats
+pub struct KeyRepeatEvent {
+    /// serial number of the event
+    pub serial: u32,
+    /// time at which the keypress occured
+    pub time: u32,
+    /// current state of the modifiers
+    pub modifiers: ModifiersState,
+    /// raw value of the key
+    pub rawkey: u32,
+    /// interpreted symbol of the key
+    pub keysym: u32,
+    /// new state of the key
+    pub state: KeyState,
+    /// utf8 interpretation of the entered text
+    ///
+    /// will always be `None` on key release events
+    pub utf8: Option<String>,
+}
+
```

Comment:
Some fields can be removed from here:

- `serial` is only meaningful for key press events, not event we generate ourselves such as these repeat events
- `state`: as this is a repeat event, this will always be `Pressed`, so it does not convey any useful information